### PR TITLE
With Eoyaml version.

### LINKED ISSUE #12: Is it possible to change the YAML library with a library that has less size?
The Simple-YAML project has 350~ kb size. I tested it with another YAML library which has 1.2 YAML 3rd edition support and the file size has been 150~ kb.
image

### PR DESCRIPTION
**Pull request by portlek**
Fixes #12 
Tests are done, the library is working correctly.

### CODE REVIEW

**Comment by portlek** [comment]
@Carleslc can you review the p.r.?

**Comment by portlek** [comment]
BTW, I changed insertion of the .yml files a little bit because the indention is no OK for eo-yaml, for example;
```yml
list:
- asd
- asd
```
this is wrong for eo-yaml and the right way is;
```yaml
list:
  - asd
  - asd
```

**Comment by Carleslc** [comment]
I personally prefer the indentation you've set for list items, but it should not be wrong to indent list items without spaces.

As per the 1.2 YAML specification the amount of indentation is a presentation detail:

https://yaml.org/spec/1.2/spec.html#id2777534

> The amount of indentation is a presentation detail and must not be used to convey content information.

And for lists, `-` character is considered as indentation.

> The “-”, “?” and “:” characters used to denote block collection entries are perceived by people to be part of the indentation.

File: src/org/simpleyaml/configuration/file/YamlConfiguration.java [hunk 1]
```diff
@@ -29,19 +25,10 @@
 
     protected static final String BLANK_CONFIG = "{}\n";
 
-    private final DumperOptions yamlOptions = new DumperOptions();
-    private final Representer yamlRepresenter = new YamlRepresenter();
-    private final Yaml yaml = new Yaml(new YamlConstructor(), yamlRepresenter, yamlOptions);
-
 	@Override
     public String saveToString() {
-        yamlOptions.setIndent(options().indent());
```
Comment (Carleslc) [inline src/org/simpleyaml/configuration/file/YamlConfiguration.java:38]: There is some way to customize indentation with eo-yaml?

Currently with SnakeYAML and Simple-YAML you can do it with:

```java
yamlFile.options().indent(4);
```

The configured indentation is then set in this line to DumperOptions.

File: test.yml [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
+    - will
+    - be
+    - in
+    - a
+    - separated
+    - entry
 math:
   pi: 3.141592653589793
 timestamp:
-  canonicalDate: 2020-07-04T13:18:04.458Z
```
Comment (Carleslc) [inline test.yml:17]: This date is not correctly saved.

Is set at `YamlTest.java`:

```java
Date now = new Date();
section.set("canonicalDate", now);
```

File: src/org/simpleyaml/utils/Helper.java [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.simpleyaml.configuration.ConfigurationSection;
+import org.simpleyaml.configuration.serialization.ConfigurationSerializable;
+import org.simpleyaml.configuration.serialization.ConfigurationSerialization;
+
+public final class Helper {
+
+    private static final List<Class<?>> SCALAR_TYPES = Arrays.asList(
```
Comment (Carleslc) [inline src/org/simpleyaml/utils/Helper.java:40]: `Date.class` can be added to represent and save dates correctly, for instance, the `timestamp.canonicalDate` in `test.yml`.

File: src/org/simpleyaml/utils/Helper.java [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+            .filter(node -> node instanceof Scalar)
+            .map(YamlNode::asScalar)
+            .forEach(keyNode ->
+                Helper.yamlNodeAsObject(mapping.value(keyNode)).ifPresent(o ->
+                    map.put(keyNode.value().replace("\"", ""), o)));
+        return map;
+    }
+
+    private static Map<String, Object> withoutMemorySection(final Map<String, Object> values) {
+        final Map<String, Object> map = new HashMap<>();
```
Comment (Carleslc) [inline src/org/simpleyaml/utils/Helper.java:74]: `HashMap` should be changed to `LinkedHashMap` to preserve set order.

File: src/org/simpleyaml/utils/Helper.java [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+        values.forEach((s, o) -> {
+            if (o instanceof ConfigurationSection) {
+                map.put(s, Helper.withoutMemorySection(((ConfigurationSection) o).getValues(false)));
+            } else if (o instanceof ConfigurationSerializable) {
+                final ConfigurationSerializable serializable = (ConfigurationSerializable) o;
+                final Map<String, Object> inner = new HashMap<>();
```
Comment (Carleslc) [inline src/org/simpleyaml/utils/Helper.java:80]: `HashMap` should be changed to `LinkedHashMap` to preserve attribute order.

File: test-comments.yml [hunk 6]
```diff
@@ -5,20 +5,20 @@
 # Test comments
 test:
   number: 5
+  boolean: true
```
Comment (Carleslc) [inline test-comments.yml:8]: The order of setting items should be preserved.

`test.boolean` is set after `test.string` in `YamlTest.java`.

**Comment by Carleslc** [comment]
Maybe we should change `Helper::loadFromString` `Yaml.createYamlInput(contents)` to `Yaml.createYamlInput(contents, true)` so the current test files indentation is still ok at loading?

https://github.com/decorators-squad/eo-yaml/wiki/Validation-of-Indentation#guessing-the-indentation-recommended-only-for-simple-yamls

**Comment by portlek** [comment]
Alright, I will check out these change requests. It can take a while.

**Comment by portlek** [comment]
I will develop EO-YAML for these change requests. until I develop the YAML library, I will close this p.r.

**Pull request closed without merging**